hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/SnapshotInfo.java

```diff
@@ -673,7 +700,8 @@ public boolean equals(Object o) {
         referencedReplicatedSize == that.referencedReplicatedSize &&
         exclusiveSize == that.exclusiveSize &&
         exclusiveReplicatedSize == that.exclusiveReplicatedSize &&
-        deepCleanedDeletedDir == that.deepCleanedDeletedDir;
+        deepCleanedDeletedDir == that.deepCleanedDeletedDir &&
+        Objects.equals(lastTransactionInfo, that.lastTransactionInfo);
```

Comment:
pre=upgrade snapshots would not have this field set. This can cause equality tests to be fail for such snapshots. Can this equality test cause regression in existing code paths due to this ?